tbtadm/controller.cpp

```diff
@@ -578,14 +578,16 @@ void tbtadm::Controller::approveAll(const fs::path& dir)
 // TODO: move to tbtadm-helper
 void tbtadm::Controller::approve(const fs::path& dir) try
 {
+    bool already_authorized = false;
+
     chdir(dir);
     m_out << "Authorizing " << dir << '\n';
 
     File authorized(authorizedFilename, File::Mode::Read);
     if (std::stoi(authorized.read()))
     {
         m_out << "Already authorized\n";
-        return;
+        already_authorized = true;
     }
 
     if (!m_once)
```

Comment:
1. All the addition to ACL should happen only if it isn't there already, isn't it?
2. Have to tested SL2? I'm pretty sure the driver blocks adding the key when the device is already approved. (Which is a good reason not to add this change but go for `add` command as the counterpart for the `remove` we already have.)

Reply:
1. Yes, here I check only for authorization. 
2. I have not tested SL2 yet.